### PR TITLE
Rework of image utils: gif resize, embeds, params

### PR DESCRIPTION
**Pull request by h-anjru**
Addresses a few issues with image processing.

- `data/img`: Processing rules moved to new YAML, `__init__.py` updated.
- `utils/image.py`: GIF resizing algorithm retooled to be (hopefully) faster; progress messages restyled.
- `cogs/image.py`: Superfluous `CommandInvokeError` embed avoided.

### CODE REVIEW

File: crimsobot/cogs/image.py [hunk 1]
```diff
@@ -33,8 +33,10 @@ async def get_image_and_embed(self, ctx: commands.Context, image: Optional[str],
                                   arg: Optional[int], title: str) -> None:
         """Get image, construct and send embed with result."""
         # process image
-        fp, img_format = await imagetools.process_image(ctx, image, effect, arg)
-        if fp is None:
+        try:
+            fp, img_format = await imagetools.process_image(ctx, image, effect, arg)
+        # if None is returned from utils/image.py: process_image(), then the user has already been informed
+        except TypeError:
```
Comment (williammck) [inline crimsobot/cogs/image.py:39]: where does the TypeError come from? and if it returns None as you mention, where are we bailing out? we still need to check for None if they don't have enough balance so we're not also sending an embed with an image that doesn't exist, no?

Comment (h-anjru) [reply crimsobot/cogs/image.py:39]: I am *not* sure why this has been behaving properly, but so far it is. The traceback if I remove the try/except:
```
ERROR: INVOKE ERROR
    guild: crimsoBOT's ship (552650672965943296)
     user: crimso#4547 (310618614497804289)
      msg: >acid 1 https://cdn.discordapp.com/attachments/554799675912355861/934735132571631646/IMG_5803.gif
        (crimsobot.bot)
Traceback (most recent call last):
  File "/home/anjru/.cache/pypoetry/virtualenvs/crimsobot-oMUi6eBF-py3.8/lib/python3.8/site-packages/discord/ext/commands/core.py", line 85, in wrapped
    ret = await coro(*args, **kwargs)
  File "/mnt/c/Users/handr/dev/crimsoBOT/crimsobot/cogs/image.py", line 68, in acid
    await self.get_image_and_embed(ctx, image, effect, number_of_hits, title)
  File "/mnt/c/Users/handr/dev/crimsoBOT/crimsobot/cogs/image.py", line 37, in get_image_and_embed
    fp, img_format = await imagetools.process_image(ctx, image, effect, arg)
TypeError: cannot unpack non-iterable NoneType object```

File: crimsobot/data/img/rules.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+image:
+  max_filesize: 8000000
+gif:
+  cost_per_frame: 0.10
+  max_frames: 300
```
[the remark this answers is not in the document]
Comment (h-anjru) [reply crimsobot/data/img/rules.yaml:5]: We don't have newlines at the end of some other YAMLs we have sitting around but I was thinking we should have them. I think I'll add it.

File: crimsobot/cogs/image.py (hunk 1 above)
Comment (h-anjru) [reply crimsobot/cogs/image.py:39]: Oh wait, it's because `process_image()` is returning `None` instead of the expected tuple, isn't it?

Comment (h-anjru) [reply crimsobot/cogs/image.py]: Addressed in [latest commit](https://github.com/crimsobot/crimsoBOT/pull/137/commits/c5a81dcc8961e26e9971ae6f59dfc05258ec3961).

**Pull request merged**